### PR TITLE
Dev: Add param conversion removal process

### PR DESCRIPTION
**Pull request by Gone4Dirt**
This PR adds some details around a new process to be implimented when adding parameter conversions.  It also updates the steps to perform a parameter conversion.

The process is simple: add a standard expression: PARAMETER_CONVERSION that we can grep for.  And detail the date and version when the conversion can be removed.  This is based on what has been disscussed on Discord and the dev call.  Happy to modify the process as maintainers think is appropriate. 

Once, we are happy with the process, I am happy to go on a hunt to find the already added conversions to add the comments appropriatley.

### CODE REVIEW

**Comment by Gone4Dirt** [comment]
@Hwurzburg When do we do the wiki Dev call?  Is it done as part of the regular dev call or is there a seperate call?

**Comment by Hwurzburg** [comment]
Regular devcall

File: dev/source/docs/code-overview-adding-a-new-parameter.rst [hunk 1]
```diff
@@ -264,10 +264,41 @@ From time to time parameters need to be altered or renamed. ArduPilot has capabi
...
+
+For example:  A conversion was added to master on the 14-Aug-2021, at which point the latest version of the stable release is 4.0.7. Therefore, the conversion can be removed after 14-Aug-2023 and once the stable release is greater than 4.2.
+
+To make it easier for to find and remove the conversions, please add the following as a comment to any parameter conversion that you add:
+
+::
+
+    /*
+    PARAMETER_CONVERSION
+    Remove when: stable release > 4.2 && date > 14-Aug-2023
```
Comment (tridge) [inline dev/source/docs/code-overview-adding-a-new-parameter.rst:301]: discussed on the dev call, and decided that adding the date the conversion code was added rather than the date to remove. The date to remove depends on how widely used a version is.

File: dev/source/docs/code-overview-adding-a-new-parameter.rst [hunk 2]
```diff
@@ -264,10 +264,41 @@ From time to time parameters need to be altered or renamed. ArduPilot has capabi
...
 
-    * Change the delay here from 1ms to 2ms [here in AP_Param::show_all](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Param/AP_Param.cpp#L2414)
-
-    * Remove the #if / #endif from [here in Copter's system.cpp](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/system.cpp#L262)
-
     * Start the old code in SITL and all the parameter names and their magic numbers will be displayed
 
     * Copy-paste from Copter's Parameters.cpp file's [existing parameter conversion tables like the ones done for the attitude controller's filters](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/Parameters.cpp#L1219).
+
+.. tip::
```
Comment (peterbarker) [inline dev/source/docs/code-overview-adding-a-new-parameter.rst:271]: Please move this tip into a separate PR.

File: dev/source/docs/code-overview-adding-a-new-parameter.rst [hunk 3]
```diff
@@ -264,10 +264,41 @@ From time to time parameters need to be altered or renamed. ArduPilot has capabi
...
+
+    If you are using xterm to open the debug terminal you may find that the list of parameters is too long for the terminal's history to find the parameter(s) you are looking for.  To resolve this, create a file called *.Xresources* in your 
+    home directory.  Using your preferred text editor add the following lines and save:
+
+    ::
+
+        XTerm.vt100.saveLines: 4096
+        Term.vt100.scrollBar: true
+
+    Then, add ``xrdb ~/.Xresources`` to the end of your *.bashrc* and save.  Finally, re-load your *.bashrc* file with the command ``source ~/.bashrc``.  Upon re-running SITL, the xterm window should now have a scroll bar and a longer history allowing you to
```
Comment (tridge) [inline dev/source/docs/code-overview-adding-a-new-parameter.rst:281]: we're better off modifying the code to save the dump to a file using AP_Filesystem, maybe "pdump.txt"

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
@Gone4Dirt still awaiting requested changes

File: dev/source/docs/code-overview-adding-a-new-parameter.rst (hunk 2 above)
Comment (Gone4Dirt) [reply dev/source/docs/code-overview-adding-a-new-parameter.rst]: removed from this PR

File: dev/source/docs/code-overview-adding-a-new-parameter.rst (hunk 1 above)
Comment (Gone4Dirt) [reply dev/source/docs/code-overview-adding-a-new-parameter.rst]: Done.  It is simply search string and date added now.

File: dev/source/docs/code-overview-adding-a-new-parameter.rst (hunk 3 above)
Comment (Gone4Dirt) [reply dev/source/docs/code-overview-adding-a-new-parameter.rst]: This tip has been removed from this PR

**Comment by Gone4Dirt** [comment]
Changes made and rebased.

**Review by tridge — APPROVED** [review]
